packages/snaps-execution-environments/src/common/lockdown/lockdown-events.ts

```diff
@@ -0,0 +1,38 @@
+// When creating a sandbox, limitation of the events from leaking
+// sensitive objects is required. This is done by overriding own properties
+// of prototypes of all existing events.
+
+/**
+ * Targeted Event objects and properties.
+ * Note: This is a map of the prototypes that inherit from Events with
+ * properties that are identified to leak sensitive objects.
+ */
+export const targetEvents: { [index: string]: any } = Object.freeze({
+  UIEvent: ['view'],
+  MutationEvent: ['relatedNode'],
+  MessageEvent: ['source'],
+  FocusEvent: ['relatedTarget'],
+  MouseEvent: ['relatedTarget', 'fromElement', 'toElement'],
+  TouchEvent: ['targetTouches', 'touches'],
+  Event: ['target', 'currentTarget', 'srcElement', 'composedPath'],
+});
+
+/**
+ * Attenuate Event objects by replacing its own properties.
+ */
+export function executeLockdownEvents() {
+  Object.keys(targetEvents).forEach((event) => {
+    const properties = targetEvents[event];
+    for (const property of properties) {
+      // @ts-expect-error Ignore index signature error.
+      if (globalThis[event]) {
+        // @ts-expect-error Ignore index signature error.
+        Object.defineProperty(globalThis[event].prototype, property, {
+          value: undefined,
+          configurable: false,
+          writable: false,
+        });
+      }
```

Comment:
This removes the need for `@ts-expect-error`.
```suggestion
      if (hasProperty(globalThis, event)) {
        Object.defineProperty(
          globalThis[event as keyof typeof globalThis].prototype,
          property,
          {
            value: undefined,
            configurable: false,
            writable: false,
          },
        );
      }
```

Reply:
Thanks! That's a good solution. Added.